### ISSUE TITLE
OWTF (run from docker image) hangs after a few plugins

### ISSUE THREAD

**Issue by smogg** [owtf/owtf#1044]
**tldr;**
Running the docker image from https://github.com/owtf/owtf-docker; the app generally hangs and never completes the full scan after a few plugins

## Expected Behavior
The app should continue scanning according to the worklist, or at least log what's happening

## Current Behavior
After selecting a bunch of plugins, some of them generate a report. At some however, everything hangs and one sees no additional output. The total scan progress is never finished (even if I remove the `workers` and the list is empty. 

![image](https://user-images.githubusercontent.com/1500714/73213033-c154bf80-414f-11ea-9c74-9ca2e9d51db0.png)

The terminal shows the last plugin's logs, and nothing is ever displayed after this

```
[-] Time=3m, 26s, 528ms
[-] __________ 4 - Target: https://demo.testfire.net -> Plugin: Session Management Schema (web/semi_passive) __________
```

## Steps to Reproduce (for bugs)
Pull the docker image and run it:

```
➜  ~ docker run -it --privileged -p 8008:8008 -p 8009:8009 -p 8010:8010 owtf/owtf /bin/bash
```

Browsing the internet through a proxied browser results in (not sure how relevant that is, even if I don't browse before running the plugins the results is the same - everything just hangs): https://pastebin.com/C0jrymKM

## Context
I was trying to run the OWTF Docker image to use the app (for the first time) on demo.testfire.net. This is the nth-time I'm trying to run the plugins on this (or many other) websites and it always stops at one point, after a few plugins. I waited multiple hours for anything to happen, no sucess.

## Your Environment
* Installation method used: docker pull owtf/owtf
* Version/branch used: MacinOWTF
* Operating System and version (like Kali, Debian, ArchLinux, etc): Docker 2.2.0.0 on MacOS

**Comment by mircealungu** [comment]
same problem when running with OS X as host.

**Comment by AkikoOrenji** [comment]
Same problem here with running on Docker Desktop latest on Windows 10 and a git pull just from yesterday.  Various other problems in addition to the hangs and plugins not working.....

Even though some tests which do complete says Successful when you click Browse everything points to http://127.0.0.1:8010/null/ 

![image](https://user-images.githubusercontent.com/30274508/112146565-d764ef80-8c2f-11eb-8cca-60b357fe2105.png)

When clicking on workers > show logs i just get some raw HTML

![image](https://user-images.githubusercontent.com/30274508/112146928-3cb8e080-8c30-11eb-9e3a-f9df072d3cd3.png)

The following error log is found on the main page:

`#### OWTF Bug Report

{"  File \"/bin/owtf/lib/python3.9/site-packages/owtf-2.6.0-py3.9.egg/owtf/workers/local.py\", line 44, in pseudo_run
    target_manager.set_target(target[\"id\"])
","  File \"/bin/owtf/lib/python3.9/site-packages/owtf-2.6.0-py3.9.egg/owtf/managers/target.py\", line 131, in set_target
    self.target_config = get_target_config_by_id(self.session, target_id)
","  File \"/bin/owtf/lib/python3.9/site-packages/owtf-2.6.0-py3.9.egg/owtf/managers/target.py\", line 362, in get_target_config_by_id
    target_obj = session.query(Target).get(id)
","  File \"/bin/owtf/lib/python3.9/site-packages/sqlalchemy/orm/query.py\", line 959, in get
    return self._get_impl(ident, loading.load_on_pk_identity)
","  File \"/bin/owtf/lib/python3.9/site-packages/sqlalchemy/orm/query.py\", line 1068, in _get_impl
    return db_load_fn(self, primary_key_identity)
","  File \"/bin/owtf/lib/python3.9/site-packages/sqlalchemy/orm/loading.py\", line 282, in load_on_pk_identity
    return q.one()
","  File \"/bin/owtf/lib/python3.9/site-packages/sqlalchemy/orm/query.py\", line 3274, in one
    ret = self.one_or_none()
","  File \"/bin/owtf/lib/python3.9/site-packages/sqlalchemy/orm/query.py\", line 3243, in one_or_none
    ret = list(self)
","  File \"/bin/owtf/lib/python3.9/site-packages/sqlalchemy/orm/query.py\", line 3316, in __iter__
    return self._execute_and_instances(context)
","  File \"/bin/owtf/lib/python3.9/site-packages/sqlalchemy/orm/query.py\", line 3341, in _execute_and_instances
    result = conn.execute(querycontext.statement, self._params)
","  File \"/bin/owtf/lib/python3.9/site-packages/sqlalchemy/engine/base.py\", line 988, in execute
    return meth(self, multiparams, params)
","  File \"/bin/owtf/lib/python3.9/site-packages/sqlalchemy/sql/elements.py\", line 287, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
","  File \"/bin/owtf/lib/python3.9/site-packages/sqlalchemy/engine/base.py\", line 1101, in _execute_clauseelement
    ret = self._execute_context(
","  File \"/bin/owtf/lib/python3.9/site-packages/sqlalchemy/engine/base.py\", line 1247, in _execute_context
    self._handle_dbapi_exception(
","  File \"/bin/owtf/lib/python3.9/site-packages/sqlalchemy/engine/base.py\", line 1466, in _handle_dbapi_exception
    util.raise_from_cause(sqlalchemy_exception, exc_info)
","  File \"/bin/owtf/lib/python3.9/site-packages/sqlalchemy/util/compat.py\", line 383, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
","  File \"/bin/owtf/lib/python3.9/site-packages/sqlalchemy/util/compat.py\", line 128, in reraise
    raise value.with_traceback(tb)
","  File \"/bin/owtf/lib/python3.9/site-packages/sqlalchemy/engine/base.py\", line 1243, in _execute_context
    self.dialect.do_execute(
","  File \"/bin/owtf/lib/python3.9/site-packages/sqlalchemy/engine/default.py\", line 552, in do_execute
    cursor.execute(statement, parameters)
"}`

How can i help the project debug these issues ?  From the docker console that's running the docker compose i can see that owtf_1 just hangs on some plugins in some instances. Other instances the plugins dont even appear to trigger any output on the console at all e.g. like something else has hung and is blocking the plugin from running. 

I've checked  and can ping from the docker container to the internet and resolve public DNS.

![image](https://user-images.githubusercontent.com/30274508/112147798-370fca80-8c31-11eb-97fb-e40e6a13960b.png)

Seems to be a few of these 

![image](https://user-images.githubusercontent.com/30274508/112147929-5d356a80-8c31-11eb-8eb9-02746aa55eb8.png)

Other plugin related errors:

![image](https://user-images.githubusercontent.com/30274508/112148225-bdc4a780-8c31-11eb-8a7d-8a9422ca3b8c.png)

![image](https://user-images.githubusercontent.com/30274508/112148390-f06ea000-8c31-11eb-9312-d047111b96b7.png)

hang at the last plugin with no futher activity after being left for hours. UI showing no scans running. 

![image](https://user-images.githubusercontent.com/30274508/112148707-4d6a5600-8c32-11eb-8b5b-113e728aee0e.png)

Based on others comments here it appears docker has been broken for some time.  Really keen to get this working so let me know if i need to turn anything else on or provide other details.

**Comment by abet-v** [comment]
Same issue look like its hanging on Plugin: Session Management Schema  I tryed to stop and re run, same issue. 
any news ?

**Comment by viyatb** [comment]
Closing this as stale. There has not been a current repro in a long time, and the code has moved enough that the original report is no longer actionable. If it still happens on current `develop`, a fresh issue with steps and logs would be more useful.
